src/test/unit/VoteWeigherBaseUnit.t.sol

```diff
@@ -0,0 +1,664 @@
+
+        // make sure a duplicate strategy exists
+        bool duplicateExists;
+        for (uint256 i = 0; i < strategiesAndWeightingMultipliers.length; i++) {
+            for (uint256 j = i + 1; j < strategiesAndWeightingMultipliers.length; j++) {
+                if (strategiesAndWeightingMultipliers[i].strategy == strategiesAndWeightingMultipliers[j].strategy) {
+                    duplicateExists = true;
+                    break;
+                }
+            }
+        }
+        cheats.assume(duplicateExists);
+
+        cheats.prank(serviceManagerOwner);
+        cheats.expectRevert("VoteWeigherBase._addStrategiesConsideredAndMultipliers: cannot add same strategy 2x");
+        voteWeigher.createQuorum(strategiesAndWeightingMultipliers);
+    }
+
+    function testCreateQuorum_EmptyStrategiesAndWeightingMultipliers_Reverts() public {
+        IVoteWeigher.StrategyAndWeightingMultiplier[] memory strategiesAndWeightingMultipliers;
+        cheats.prank(serviceManagerOwner);
+        cheats.expectRevert("VoteWeigherBase._addStrategiesConsideredAndMultipliers: no strategies provided");
+        voteWeigher.createQuorum(strategiesAndWeightingMultipliers);
+    }
+
+    function testCreateQuorum_StrategiesAndWeightingMultipliers_WithZeroWeight(
+        IVoteWeigher.StrategyAndWeightingMultiplier[] memory strategiesAndWeightingMultipliers
+    ) public {
+        strategiesAndWeightingMultipliers = _removeDuplicates(strategiesAndWeightingMultipliers);
+        cheats.assume(strategiesAndWeightingMultipliers.length <= voteWeigher.getMaxWeighingFunctionLength());
+        cheats.assume(strategiesAndWeightingMultipliers.length > 0);
+        // make sure a zero weight exists
+        bool zeroWeightExists;
+        for (uint256 i = 0; i < strategiesAndWeightingMultipliers.length; i++) {
+            if (strategiesAndWeightingMultipliers[i].multiplier == 0) {
+                zeroWeightExists = true;
+                break;
+            }
+        }
+        cheats.assume(zeroWeightExists);
+
+        cheats.prank(serviceManagerOwner);
+        cheats.expectRevert("VoteWeigherBase._addStrategiesConsideredAndMultipliers: cannot add strategy with zero weight");
+        voteWeigher.createQuorum(strategiesAndWeightingMultipliers);
+    }
+
+    function testCreateQuorum_MoreThan256Quorums_Reverts(
+        IVoteWeigher.StrategyAndWeightingMultiplier[] memory strategiesAndWeightingMultipliers
+    ) public {
+        strategiesAndWeightingMultipliers = _convertToValidStrategiesAndWeightingMultipliers(strategiesAndWeightingMultipliers);
+            
+        cheats.startPrank(serviceManagerOwner);
+        for (uint i = 0; i < 255; i++) {
+            voteWeigher.createQuorum(strategiesAndWeightingMultipliers);
+        }
+        cheats.expectRevert("VoteWeigherBase._createQuorum: number of quorums cannot 256");
+        voteWeigher.createQuorum(strategiesAndWeightingMultipliers); 
+    }
```

Comment:
this test needs to be corrected to actually check for 256 quorums, rather than 255 (incorrect test matches incorrect implementation).
maybe add a line that checks that the number of quorums *is actually 256* before the `expectRevert`?